softvis3d-frontend/src/components/ui/selectbox/SelectBox.tsx

```diff
@@ -67,19 +77,39 @@ export default class SelectBox extends React.Component<SelectBoxProps, any> {
         );
     }
 
+    private addValue(v: SelectOptionValue) {
+        this.values[v.getValue()] = v;
+    }
+
+    private clearValues() {
+        this.values = {};
+    }
+
     private renderChildren(): Array<React.Component<any, any>> {
+        this.clearValues();
+        const ref = (o: SelectOption | null) => {
+            if (!o) {
+                this.clearValues();
+            } else {
+                const v: SelectOptionValue = o.props.value;
+                this.addValue(v);
+            }
+        };
+
         return React.Children.map<any>(
             (this.props.children as Array<SelectOption|SelectGroup>),
             (child: React.ReactElement<any>) => {
                 if (child.type === SelectOption) {
                     return React.cloneElement(child, {
                         checked: child.props.value === this.props.value,
-                        disabled: this.props.disabled || child.props.disabled
+                        disabled: this.props.disabled || child.props.disabled,
+                        ref
                     });
                 } else if (child.type === SelectGroup) {
                     return React.cloneElement(child, {
                         selectedValue: this.props.value,
-                        disabled: this.props.disabled || child.props.disabled
+                        disabled: this.props.disabled || child.props.disabled,
+                        optionRef: ref
                     });
                 } else {
                     return child;
```

Comment:
How to deal with that. As far as i understand this code "should never be called" but still we have to write some code here to provide a return value. Do you know a way around this? I have a similar issue in the file StatusActionQueue.tsx - line 45 which leads to the test error output mentioned by you in #60 (Attempt to read an array index (0) that is out of bounds (0). Please check length first. Out of bound indices will not be tracked by MobX). I think these are similar issues.